### PR TITLE
Handle the situation when an attempt to bind a socket to a specific I…

### PR DESCRIPTION
**Pull request by rawafb**
…P address and port fails during startup. Provide an error message for the user.

### CODE REVIEW

File: src/emcgw/server.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import socket
 from .logger import logger
+import logging
```
Comment (rohzb) [inline src/emcgw/server.py:3]: do not `import logging`, use `logger `above - that way we can use command line options for extra verbosity.

File: src/emcgw/server.py [hunk 2]
```diff
@@ -133,7 +134,14 @@ def start(self):
         """
         server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        server_socket.bind((self.local_host, self.local_port))
+        
+        try:
+            server_socket.bind((self.local_host, self.local_port))
+        except socket.error as e:
+            logging.error(f"Failed to bind to {self.local_host}:{self.local_port} - {e}")
```
Comment (rohzb) [inline src/emcgw/server.py:141]: same here: it should be `logger `instead of `logging`

**Review by rohzb — CHANGES_REQUESTED** [review]
Cool!

1 change is required - use `logger` already imported instead importing logging. It sets extra verbosity levels and allow enable/disable printing of them over command line options - see comments to particular line

File: src/emcgw/server.py (hunk 1 above)
Comment (rawafb) [reply src/emcgw/server.py]: Done!

**Pull request merged**